include/kernel.h

```diff
@@ -90,7 +90,11 @@ extern "C" {
 #define K_HIGHEST_APPLICATION_THREAD_PRIO (K_HIGHEST_THREAD_PRIO)
 #define K_LOWEST_APPLICATION_THREAD_PRIO (K_LOWEST_THREAD_PRIO - 1)
 
-typedef sys_dlist_t _wait_q_t;
+typedef struct {
+	sys_dlist_t waitq;
+} _wait_q_t;
+
+#define _WAIT_Q_INIT(wait_q) { SYS_DLIST_STATIC_INIT(&(wait_q)->waitq) }
```

Comment:
Just curious but why the abstraction?

Reply:
Because (c.f. the scheduler rewrite PR, underneath which this needs to sit) I'm about to rip out _wait_q_t and replace it with a switchable implementation that can be either a dlist or a rbtree, and I got sick of stumbling over all the spots where we were promiscuously assuming that it was a dlist.  So now it's an opaque type that has to have its own API (which I can conveniently replace in one targetted swoop).